currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/BitcoinFragment.kt

```diff
@@ -99,6 +94,35 @@ class BitcoinFragment : BaseFragment(R.layout.fragment_bitcoin),
         }
     }
 
+    private fun handleWalletNavigation() {
+        val navController = findNavController()
+
+        val vWalletFileMainNet = File(
+            this.requireContext().applicationContext.filesDir,
+            "$MAIN_NET_WALLET_NAME.wallet"
+        )
+
+        val vWalletFileTestNet = File(
+            this.requireContext().applicationContext.filesDir,
+            "$TEST_NET_WALLET_NAME.wallet"
+        )
+
+        if ((vWalletFileMainNet.exists() && vWalletFileMainNet.exists()) && !WalletManagerAndroid.isInitialized()) {
+            navController.navigate(R.id.daoLoginChoice)
+        } else if ((vWalletFileMainNet.exists() || vWalletFileTestNet.exists()) && !WalletManagerAndroid.isInitialized()) {
+            val params = when (vWalletFileTestNet.exists()) {
+                true -> BitcoinNetworkOptions.TEST_NET
+                false -> BitcoinNetworkOptions.PRODUCTION
+            }
+            val config = WalletManagerConfiguration(params)
+            WalletManagerAndroid.Factory(this.requireContext().applicationContext)
+                .setConfiguration(config).init()
+            navController.navigate(R.id.blockchainDownloadFragment)
+        } else if (!vWalletFileMainNet.exists() && !vWalletFileTestNet.exists()) {
+            navController.navigate(R.id.daoLoginChoice)
+        }
```

Comment:
For this part it might be nice to extract the ```vWalletFileMainNet.exists()``` to a boolean to improve readability. 
Also, comments would be nice to explain why these navigation paths are taken, without having the read the long if statements.